toolkit/authentication/src/test/java/com/arcgismaps/toolkit/authentication/SafeSuspendableCancellableTest.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ *
+ *  Copyright 2025 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.arcgismaps.toolkit.authentication
+
+import com.arcgismaps.toolkit.authentication.utils.safeSuspendableCancellable
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.async
+import kotlinx.coroutines.test.runTest
+import org.junit.Ignore
+import org.junit.Test
+import java.util.concurrent.atomic.AtomicBoolean
+
+class SafeSuspendableCancellableTest {
```

Comment:
Missing class level doc:
```suggestion
/**
 * Tests for [safeSuspendableCancellable].
 * @since 200.8.1
 */
class SafeSuspendableCancellableTest {
```